plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/completion/SQLQueryCompletionContext.java

```diff
@@ -342,8 +344,16 @@ private List<SQLQueryCompletionItem> prepareObjectComponentCompletions(
                         components = Stream.empty();
                     }
 
-                    return components.filter(a -> (componentType == null || componentType.isInstance(a)) && a.getName().toLowerCase().contains(componentNamePart))
-                        .map(SQLQueryCompletionItem::forDbObject)
+                    return components.filter(a -> (componentType == null || componentType.isInstance(a))
+                            && a.getName().toLowerCase().contains(componentNamePart))
+                        .map(x -> {
+                            //todo transform to the map strategy if add one more case
+                            if (x instanceof DBSEntity dbsEntity) {
+                                return SQLQueryCompletionItem.forRealTable(dbsEntity, true);
```

Comment:
We can't just put true to the second argument. We need to check if it's a first usage of the table or not. Use `alreadyReferencedTables` set for that